### PR TITLE
Search for RNA cases

### LINKED ISSUE #3646: Search for cases with RNA data
It would be nifty to be able to spot in the list, and search for cases where RNA data (splice junction tracks) have been loaded. Perhaps an additional badge in the `Sequencing` column, plus a toggle or search field option?

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix #3646 --> Filter cases for cases with RNA data

Adds a checkbox on cases filter to filter for cases using these conditions:

```
EXISTS_NOT_NULL = {"$exists": True, "$nin": [None, ""]}
query = {
   "owner": owner,
    "$or": [
                {"gene_fusion_report": EXISTS_NOT_NULL},
                {"individuals.splice_junctions_bed": EXISTS_NOT_NULL},
                {"individuals.rna_coverage_bigwig": EXISTS_NOT_NULL},
            ],
 }
```

These conditions might eventually be expanded when we add new RNA-seq data, for instance RNA multiqc etc

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the paxa software installed on `hasta`:
 - Log in into hasta: `ssh <USER.NAME>@hasta.scilifelab.se`
 - Activate the staging environment with the command `us` (Use Stage)
 - Run the command `paxa` and follow the instructions. Make sure to specify scout-stage as the resource you request and the server cg-vm1 as server.
1. Log out from the hasta server.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, log out from `cg-vm1` and log in again in the `hasta` server, repeat the `hasta` and `paxa` procedure, which will release the allocated resource (scout-stage) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR, DN

### CODE REVIEW

**Comment by northwestwitch** [comment]
Tested on stage with cust003: returns 5 cases with RNA-seq data

File: scout/adapter/mongo/case.py [hunk 1]
```diff
@@ -307,56 +309,93 @@ def cases(
                 instead returns corresponding query dict
                 that can be reused in compound queries or for testing.
         """
+
+        def _set_query_value(query, value, set_key, set_value):
```
[the remark this answers is not in the document]
Comment (dnil) [reply scout/adapter/mongo/case.py:313]:
```suggestion
        def _conditional_set_query_value(query, condition, set_key, set_value):
```
It works just fine as is, but value and set_value may sound a bit similar next month?

File: tests/adapter/mongo/test_case_handling.py [hunk 2]
```diff
@@ -757,6 +757,19 @@ def test_verification_missing_cases(real_adapter, case_obj, user_obj, institute_
     assert len(adapter.verification_missing_cases(institute_obj["_id"])) == 1
 
 
+def test_rna_cases(real_adapter, case_obj):
```
Comment (dnil) [inline tests/adapter/mongo/test_case_handling.py:760]: 💯

**Review by dnil — APPROVED** [review]

**Comment by northwestwitch** [comment]
Thanks for the review @dnil, merging!

**Pull request merged**